src/CoreBundle/Tests/Response/ResponseVariableReplacerTest.php

```diff
@@ -0,0 +1,173 @@
+                [
+                    'two' => 'nothing',
+                    'placeholders' => 'to see',
+                ],
+                'This is my string with nothing to see and I am proud of it.',
+            ],
+            'string-with-unknown-placeholder' => [
+                'This is my string with a [{valid-placeholder}] and an [{unknown-placeholder}] and I am proud of it.',
+                [
+                    'valid-placeholder' => 'very meaningful text',
+                ],
+                'This is my string with a very meaningful text and an [{unknown-placeholder}] and I am proud of it.',
+            ],
+            'string-with-authenticity-token' => [
+                'This is my string with [{cmsauthenticitytoken}] and I am proud of it.',
+                [],
+                'This is my string with such-token and I am proud of it.',
+            ],
+            'array-with-placeholder' => [
+                [
+                    'such-key' => 'many-value',
+                    'so-placeholder' => '<<< [{wow}] xxx',
+                ],
+                [
+                    'wow' => 'woohooo',
+                ],
+                [
+                    'such-key' => 'many-value',
+                    'so-placeholder' => '<<< woohooo xxx',
+                ],
+            ],
+            'object-with-placeholder' => [
+                $testObject,
+                $placeholders,
+                $expectedResultObject,
+            ],
+        ];
+    }
+
+    private function givenResponseVariableReplacer(array $placeholders): void
+    {
+        $this->authenticityTokenManagerMock = $this->prophesize(AuthenticityTokenManagerInterface::class);
+        $this->authenticityTokenManagerMock->addTokenToForms(Argument::any())->will(function (array $args) {
+            return $args[0];
+        });
+        $this->authenticityTokenManagerMock->getStoredToken()->willReturn('such-token');
+        $this->flashMessageServiceMock = $this->prophesize(FlashMessageServiceInterface::class);
+        $this->flashMessageServiceMock->injectMessageIntoString(Argument::any())->will(function (array $args) {
+            return $args[0];
+        });
+        $this->subject = new ResponseVariableReplacer(
+            $this->authenticityTokenManagerMock->reveal(),
+            $this->flashMessageServiceMock->reveal()
+        );
+
+        foreach ($placeholders as $key => $value) {
+            $this->subject->addVariable($key, $value);
+        }
+    }
+
+    /**
+     * @param object|array|string $input
+     */
+    private function whenReplaceVariablesIsCalled($input): void
+    {
+        $this->actualResult = $this->subject->replaceVariables($input);
+    }
+
+    private function thenTheExpectedResultShouldBeReturned($expectedResult): void
+    {
+        $this->assertEquals($expectedResult, $this->actualResult);
+    }
+
+    public function testTokenInjectionFailedException(): void
```

Comment:
Please move this test below the other one so one can see all the tests side by side.

Reply:
done (moved it under the data provider of the other test, as test and provider shouldn't be separated). I think we should discuss the desired order again when we find the time.